File: app/src/main/java/com/serenegiant/libcommon/CameraSurfaceFragment.java
package com.serenegiant.libcommon;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.serenegiant.widget.CameraSurfaceView;

import org.jetbrains.annotations.NotNull;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * SurfaceViewへカメラ映像を流し込んで表示するだけのFragment
 */
public class CameraSurfaceFragment extends BaseFragment {
	private static final boolean DEBUG = true;	// set false on production
	private static final String TAG = CameraSurfaceFragment.class.getSimpleName();

	private CameraSurfaceView mCameraView;

	public CameraSurfaceFragment() {
		super();
		// デフォルトコンストラクタが必要
	}

	@Override
	public void onAttach(@NotNull final Context context) {
		super.onAttach(context);
		requireActivity().setTitle(TAG);
	}

	@Nullable
	@Override
	public View onCreateView(
		@NonNull final LayoutInflater inflater,
		@Nullable final ViewGroup container,
		@Nullable final Bundle savedInstanceState) {

		if (DEBUG) Log.v(TAG, "onCreateView:");
		final View rootView = inflater.inflate(
			R.layout.fragment_camera_surfaceview,
			container, false);
		mCameraView = rootView.findViewById(R.id.cameraView);
		return rootView;
	}

	@Override
	protected void internalOnResume() {
		super.internalOnResume();
		if (DEBUG) Log.v(TAG, "internalOnResume:");
		mCameraView.onResume();
	}

	@Override
	protected void internalOnPause() {
		if (DEBUG) Log.v(TAG, "internalOnPause:");
		mCameraView.onPause();
		super.internalOnPause();
	}
}

File: app/src/main/java/com/serenegiant/libcommon/CameraSurfaceFragment.kt
package com.serenegiant.libcommon
/*
 * libcommon
 * utility/helper classes for myself
 *
 * Copyright (c) 2014-2025 saki t_saki@serenegiant.com
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
*/

import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.serenegiant.widget.CameraSurfaceView

class CameraSurfaceFragment : BaseFragment() {

	private var mCameraView: CameraSurfaceView? = null

	override fun onAttach(context: Context) {
		super.onAttach(context)
		requireActivity().title = TAG
	}

	override fun onCreateView(
		inflater: LayoutInflater,
		container: ViewGroup?,
		savedInstanceState: Bundle?): View? {
		if (DEBUG) Log.v(TAG, "onCreateView:")
		val rootView = inflater.inflate(
			R.layout.fragment_camera_surfaceview,
			container, false)
		mCameraView = rootView.findViewById(R.id.cameraView)
		return rootView
	}

	override fun internalOnResume() {
		super.internalOnResume()
		if (DEBUG) Log.v(TAG, "internalOnResume:")
		mCameraView!!.onResume()
	}

	override fun internalOnPause() {
		if (DEBUG) Log.v(TAG, "internalOnPause:")
		mCameraView!!.onPause()
		super.internalOnPause()
	}

	companion object {
		private const val DEBUG = true // TODO set false on release
		private val TAG = CameraSurfaceFragment::class.java.simpleName

		fun newInstance() = CameraSurfaceFragment().apply {
			arguments = Bundle().apply {
				// 今は何もない
			}
		}
	}

}
